buildkite/resource_pipeline_schedule.go

```diff
@@ -117,17 +118,37 @@ func (ps *pipelineSchedule) Create(ctx context.Context, req resource.CreateReque
 
 	log.Printf("Creating Pipeline schedule %s ...", plan.Label.ValueString())
 
-	envVars := envVarsMapFromTfToString(ctx, plan.Env)
-	apiResponse, err := createPipelineSchedule(ctx,
-		ps.client.genqlient,
-		plan.PipelineId.ValueString(),
-		plan.Label.ValueStringPointer(),
-		plan.Cronline.ValueStringPointer(),
-		plan.Message.ValueStringPointer(),
-		plan.Commit.ValueStringPointer(),
-		plan.Branch.ValueStringPointer(),
-		&envVars,
-		plan.Enabled.ValueBool())
+	timeouts, diags := ps.client.timeouts.Create(ctx, DefaultTimeout)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var apiResponse *createPipelineScheduleResponse
+	err := retry.RetryContext(ctx, timeouts, func() *retry.RetryError {
+		var err error
+
+		envVars := envVarsMapFromTfToString(ctx, plan.Env)
+		apiResponse, err = createPipelineSchedule(ctx,
+			ps.client.genqlient,
+			plan.PipelineId.ValueString(),
+			plan.Label.ValueStringPointer(),
+			plan.Cronline.ValueStringPointer(),
+			plan.Message.ValueStringPointer(),
+			plan.Commit.ValueStringPointer(),
+			plan.Branch.ValueStringPointer(),
+			&envVars,
+			plan.Enabled.ValueBool())
+
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(err)
```

Comment:
I think this could be changed - and you'd not need the `err` on the `RetryContext`

```suggestion
			resp.Diagnostics.AddError(
			    "Unable to create Pipeline schedule",
			    fmt.Sprintf("Unable to create Pipeline schedule: %s", err.Error()),
		    )
		    return retry.NonRetryableError(err)
```

Reply:
I am using `err` outside the code block of RetryContext.  The Create operation exits when an error is encountered on retry with this code block:

```	
       if err != nil {
		resp.Diagnostics.AddError(
			"Unable to create Pipeline schedule",
			fmt.Sprintf("Unable to create Pipeline schedule: %s", err.Error()),
		)
		return
	}
```